src/simulation/thermal/sensorThermal/_UnitTest/test_sensorThermalFaults.py

```diff
@@ -0,0 +1,216 @@
+
+    #
+    # log data
+    #
+
+    # log the RW temperature
+    tempLog = sensorThermalModel.temperatureOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, tempLog)
+
+    #
+    #   initialize Simulation
+    #
+    # Truth Values
+    if tempFault == "TEMP_FAULT_NOMINAL":
+        truthValue = -4.251289338192501
+        unitTestSim.InitializeSimulation()
+        unitTestSim.ConfigureStopTime(simulationTime)
+        unitTestSim.ExecuteSimulation()
+
+    elif tempFault == "TEMP_FAULT_STUCK_CURRENT":
+        truthValue = -2.1722164230619447
+        unitTestSim.InitializeSimulation()
+        unitTestSim.ConfigureStopTime(round(simulationTime/2))
+        unitTestSim.ExecuteSimulation()
+        sensorThermalModel.faultState = sensorThermal.TEMP_FAULT_STUCK_CURRENT
+        unitTestSim.ConfigureStopTime(simulationTime)
+        unitTestSim.ExecuteSimulation()
+
+    elif tempFault == "TEMP_FAULT_STUCK_VALUE":
+        truthValue = 10.0
+        sensorThermalModel.stuckValue = 10.0
+        unitTestSim.InitializeSimulation()
+        unitTestSim.ConfigureStopTime(round(simulationTime/2.0))
+        unitTestSim.ExecuteSimulation()
+        sensorThermalModel.faultState = sensorThermal.TEMP_FAULT_STUCK_VALUE
+        unitTestSim.ConfigureStopTime(simulationTime)
+        unitTestSim.ExecuteSimulation()
+    elif tempFault == "TEMP_FAULT_SPIKING":
+        sensorThermalModel.faultState = sensorThermal.TEMP_FAULT_SPIKING
+        truthValue = 0.1
+        unitTestSim.InitializeSimulation()
+        unitTestSim.ConfigureStopTime(simulationTime)
+        unitTestSim.ExecuteSimulation()
+        nominals = np.array([ 0.,         -0.04439869, -0.08875756, -0.13307667, -0.17735608, -0.22159584, -0.265796,   -0.30995662, -0.35407775, -0.39815946, -0.44220178, -0.48620479, -0.53016852, -0.57409304, -0.6179784,  -0.66182465, -0.70563184, -0.74940004, -0.79312929, -0.83681965, -0.88047117, -0.9240839,  -0.9676579,  -1.01119321, -1.0546899,  -1.09814802, -1.14156761, -1.18494873, -1.22829144, -1.27159578, -1.31486181, -1.35808958, -1.40127914, -1.44443055, -1.48754385, -1.5306191, -1.57365634, -1.61665564, -1.65961704, -1.7025406,  -1.74542636, -1.78827437, -1.83108469, -1.87385737, -1.91659246, -1.95929001, -2.00195007, -2.04457269, -2.08715792, -2.12970582, -2.17221642, -2.21468979, -2.25712597, -2.29952502, -2.34188697, -2.38421189, -2.42649982, -2.46875082, -2.51096492, -2.55314219, -2.59528266, -2.6373864,  -2.67945344, -2.72148384, -2.76347765, -2.80543491, -2.84735568, -2.88924,    -2.93108792, -2.9728995 , -3.01467477, -3.05641378, -3.0981166,  -3.13978325, -3.1814138,  -3.22300829, -3.26456677, -3.30608928, -3.34757587, -3.38902659, -3.43044149, -3.47182062, -3.51316402, -3.55447174, -3.59574383, -3.63698033, -3.67818129, -3.71934677, -3.76047679, -3.80157142, -3.8426307,  -3.88365467, -3.92464338, -3.96559689, -4.00651522, -4.04739844, -4.08824658, -4.1290597,  -4.16983783, -4.21058103, -4.25128934])
```

Comment:
:small_red_triangle_down: Reformat this expression so that there is a line break at every 120 character. As it is I have to scroll a long way across may screen to view all of this list. A really easy way to do this in PyCharmis to highlight the line and select Code >> Reformat Code.